### ISSUE TITLE
Accessible Text/Discernable Text Violations in noscript Tags

### ISSUE THREAD

**Issue by Cassidie** [dequelabs/axe-core#1648]
Expectation: 
Elements in a noscript tag should not be tested for accessibility.

Actual: 
Links and buttons in noscript tags are being flagged as violating accessible/ discernible text.

Motivation: 
Please allow for noscript tags to be ignored

<pre><code>
axe-core version: 3.8.0
axe-webdriver, extension or other integration version: chrome (v74) dev-tools extension

Browser and Assistive Technology versions

For Tooling issues:
- Node version: 10.15.3
- Platform:   Windows 
</code></pre>

**Comment by straker** [comment]
Thanks for the issue. Unfortunately I'm not able to reproduce this. Would you have some time to setup a failing test case I could look at?

P.s. here's the code I used.

```html
<body>
  <main>
    <h1>Hello</h1>
    <noscript>
      <a href="https://www.mozilla.com/">External Link</a>
    </noscript>
  </main>
</body>
```

**Comment by dylanb** [comment]
I also ran it with

```
<body>
  <main>
    <h1>Hello</h1>
    <noscript>
      <a href="https://www.mozilla.com/"></a>
    </noscript>
  </main>
</body>
```

Where there is an error in the `<noscript>` content and it did not fail

**Comment by Cassidie** [comment]
I apologize for not having given an example to begin with. I was testing a public url, `www.estatesales.net`, with the axe chrome extension. If you run it on the landing page of the site, you'll see the violation. The link and button reside in a custom element inside a noscript tag. I don't know if that helps or not. We assumed it was the noscript tag, because we aren't having any issues with other buttons/links inside custom elements. Any help is much appreciated!

**Comment by straker** [comment]
Thanks for the link, that was very helpful. I think I found the problem. 

Normally when you have a `<noscript>`, the browser will interpret everything inside it as text. So doing 

```html
<noscript>
  <a href="https://www.mozilla.com/"></a>
</noscript>
```

Will result in a text string of `<a href="https://www.mozilla.com/"></a>` inside the `<noscript>` tag, which is fine and expected.

However, somehow on the site the contents of the `<noscript>` tag are actual DOM elements and not just text content. This causes the elements to be evaluated for all axe-core rules. Because the `<noscript>` tag still hides the child elements, the anchor in question is hidden so fails the `link-name` rule. I would hazard a guess that the custom element is using `appendChild` to the `<noscript>` tag instead of `innerHTML`, which causes actual DOM elements to be added as children. 

As such, I don't think this is a bug in axe-core as it's correctly identifying a problem with the current state of the DOM tree.

**Comment by Cassidie** [comment]
Thank you so much! That all makes perfect sense. I will see what we can do to fix this on our end. I really, really appreciate your help looking into this. Great tool, btw! Thanks!

**Comment by kevinphelps** [comment]
I filed an issue with Angular for this: angular/angular#31315.
